### DISCUSSION TITLE
Board completely routed - 2022/08/18

### DISCUSSION THREAD

**Discussion by josuah** [tinyvision-ai-inc/pico-ice#9] [Design Review]
https://cdn.discordapp.com/attachments/985342148884987934/1009574670934229083/pico-ice_layout.pdf

**Comment by josuah** [comment]
**Ground layer barely used for signal, +3v3 much more used**
This is likely a feature if ground can be considered as part of the signal: that way, the vias permit the current to move freely on the entire uninterrupted ground plane.

**Reply by vr2045** [reply]
I like to use a solid ground plane. All signals & power are referred to a solid sheet, significantly reducing the chances of strange issues.

**Reply by josuah** [reply]
That teaches me something about grounding. 4-layers PCBs are really nice

**Comment by josuah** [comment]
**No separate "analog" Ground pin**

On the "pico" board:
[![2022-08-18-214741_1443x138_scrot](https://user-images.githubusercontent.com/55351021/185482032-5cd31579-3abc-495f-bf97-7f4ddf0fcd02.png)](https://datasheets.raspberrypi.com/pico/pico-datasheet.pdf#page=9)

But looking at the pico board itself:
![2022-08-18-214649_1357x985_scrot](https://user-images.githubusercontent.com/55351021/185483230-89218249-af51-4246-ac62-a13a82faaaab.png)

The "AGND" pin seems bound to the same global ground pad as all the others.
Maybe it is only used for the passives used for the reference voltage...
Or maybe it is there for promoting good practices.

The RP2040 chip does not seem to have any separate ground plane than the big pad.
We are probably good.

**Reply by vr2045** [reply]
Yes, they seem to have tied off the ground to a common (and split) ground plane. I think a solid ground will help. This will likely cause some issues for the ADC by adding noise. Will have to wait and see if this really affects the ADC...

**Comment by josuah** [comment]


**Reply by josuah** [reply]
My bad, these are the RESISTORS for the LED, not the LED itself...
![2022-08-18-225314_559x384_scrot](https://user-images.githubusercontent.com/55351021/185492750-54167456-f871-4c71-9379-37a94ac3abe3.png)

**Comment by josuah** [comment]
**Part reuse for USB**
The USB standard permits two 10k resistor to be put in parallel (5.0k) instead of the 5.1k, given the error margin of a few percents in the spec.
This would permit to kill the 5.1k off the BOM.
![2022-08-18-223757_1038x673_scrot](https://user-images.githubusercontent.com/55351021/185490167-08530fac-c13a-4a81-842f-e3d20e6a830c.png)

**Reply by josuah** [reply]
Given the board situation above, there would be a via needed for the two 10k in parallel, or 4 * 10k resistors in total (2 for CC1, 2 for CC2).

A via would not hurt the CC pin a lot: not a signal pin: mostly constant if not used.

**Reply by vr2045** [reply]
Good idea to look into DFM at this stage. 
In this particular case, resistor cost is far smaller than placement cost or that of an additional part to be managed given that its a common one. So given a choice, I would replace multiple resistors with a single one, not vice-versa.

**Comment by josuah** [comment]
**Overview of the JTAG header**

~Is it meant to have something soldered in there? Anything in mind?~ Yes! As seen below.
![2022-08-18-224848_722x611_scrot](https://user-images.githubusercontent.com/55351021/185492135-00ff5620-6964-4216-b653-b74ecb55a9d1.png)

**Reply by josuah** [reply]
![2022-08-18-231301_401x328_scrot](https://user-images.githubusercontent.com/55351021/185496215-8a83fcd6-294d-42dd-829f-f4569fc10f21.png)
Here goes the header. It could be a bit inconvenient in breadboard mode: bottom of the board... But who needs breadboard when you have PMODs!

**Reply by josuah** [reply]
It would be handy to have a link toward a cable for sell matching the header.
And a clear pinout:

```
│o o              o o │
│o o              o o │
│o o < BACKSIDE > o o │
│o o              o o │  ┌────────────────────────────┐
│o o              o o │ /│                            │
│o o              o o │/ │                            │
│  _                  /  │    GND  x  GND GND 3V3     │
│────'┬────┬ ┌───────┐│  │ ┌───│───│───│───│───│───┐  │
│BOOT ┤°   ├ │┌┴┴┴┴└┐││  │ │   o   o   o   o   o   │  │
│o    ┤    ├ ││:::::│││  │ │                       │  │
│o    ┴────┴ │└┬┬┬┬┬┘││  │ │   o   o   o   o   o   │  │
│-      U2   └───────┘│  │ └───│───│───│───│───│───┘  │
│      O       O      \  │    RST  x   x  SWC SWD     │
│ _   PICO-ICE v1   _ │\ │                            │
│(_)   O       O   (_)│ \│                            │
└─────────────────────┘  └────────────────────────────┘
```

^ is this correct?

**Reply by josuah** [reply]
> iCE40 FPGAs do not have a JTAG interface -- [https://stackoverflow.com/questions/49840479/lattice-ice40-jtag#49844398](Claire Clifford)

And maybe along with a note that the JTAG is only for the RP2040.

**Reply by vr2045** [reply]
Right, you got it! This connector is pretty standard and compact. Anyone doing work with embedded processors has likely got one of these lying around or has access to one so I selected this.
The RPi foundation made up a smaller connector but I dont see that as a standard yet, it just might become prevalent in which case, I'll replace this. This connector wont be installed by default.

**Comment by josuah** [comment]
**BOM overview**
Filtering components with 3 or less instances, we have mostly connectors or chips, and a few usual passives (marked with `*`, with `S` for solved):
```
1       :SW_Push_1 PTS820 J25K SMTR LFS (SW2)
1       Connector:Conn_ARM_JTAG_SWD_10 Conn_ARM_JTAG_SWD_10 (J5)
1       Connector:USB_C_Receptacle_USB2.0 USB_C_Receptacle_USB2.0 (J1)
1       Connector_Generic:Conn_01x02 BOOT_SEL (J9)
1       Device:Crystal_GND24 12 MHz (Y1)
1       Regulator_Linear:NCP1117-3.3_SOT223 NCP1117ST33T3G (U1)
1       Switch:SW_Push PTS820 J25K SMTR LFS (SW1)
1       UPduino_v3.0:AP2127K-1.2 CL9293A12L5M (U4)
1       UPduino_v3.0:Connector_TestPoint TestPoint (TP1)
1       UPduino_v3.0:Device_LED_ALT LTST-C191KGKT (D3)
1  * S  UPduino_v3.0:Device_R 100 (R11)
1       UPduino_v3.0:Device_R MF-FSMF035X-2 (PTC1)
1       UPduino_v3.0:Device_R OPEN (SJ6)
1  *    UPduino_v3.0:Diode_1N4148W 1N4148WT-7 (D1)
1       UPduino_v3.0:ICE40UP5K-SG48ITR-FPGA_Lattice ICE40UP5K-SG48ITR (U6)
1       UPduino_v3.0:W25Q32JVSS-Memory_Flash APS6404L-3SQN-SN (U5)
1       pico-ice_minimal:MCU_RaspberryPi_RP2040_RP2040 RP2040 (U3)
2       2x3FemaleHeader:2x3_2.54mm_Female 2x3_2.54mm_Female (J10, J11)
2  * S  Device:C 1u (C6, C11)
2  * S  Device:C 27p (C1, C2)
2  * S  Device:R 1k (R1, R2)
2       UPduino_v3.0:Device_LED_ARGB MHPA1010RGBDT (D2, D4)
2  * S  UPduino_v3.0:Device_R 5.1K (R10, R24)
2       UPduino_v3.0:W25Q32JVSS-Memory_Flash W25Q32JVSSIQ (U2, U8)
3       UPduino_v3.0:Device_R SHORT (SJ1, SJ4, SJ5)
```

P.S.: no need to read everything below, the BOM looks good.

**Reply by josuah** [reply]
# R11 (solved)

![2022-08-19-110411_469x131_scrot](https://user-images.githubusercontent.com/55351021/185584882-230588cb-5a7f-4d66-8763-4f3fc073e933.png)

As written in the ICE40 sysCLOCK PLL guide: https://www.latticesemi.com/-/media/LatticeSemi/Documents/ApplicationNotes/IK2/FPGA-TN-02052-1-3-iCE40-sysCLOCK-PLL-Design-Usage-Guide.ashx
![2022-08-19-110618_693x608_scrot](https://user-images.githubusercontent.com/55351021/185585378-c53d5592-d225-4d81-ba62-7c0566e3f2d6.png)

As I understand it, there should be some "GNDPLL" pin on the ICE40 but...

![2022-08-19-112036_1468x765_scrot](https://user-images.githubusercontent.com/55351021/185587845-b227e53f-0c29-4a02-b5ff-5d8a4fdbc75d.png)

No trace of GNDPLL in https://www.latticesemi.com/view_document?document_id=51971
![2022-08-19-113336_733x194_scrot](https://user-images.githubusercontent.com/55351021/185593918-58f3a2d8-d1f5-4e5d-85e7-99956528b033.png)

Confirmed by EEVblog:
> The trick is the ice40UP5k in SG48 packages does not possess the GNDPLL pin. The appnote says the PLL's decoupling filter shall be wired between VCCPLL and GNDPLL, and "GNDPLL must not be connected to the board's ground". They want to minimize the noise getting into the PLL, therefore in other packages the special GNDPLL pin is directly connected to the internal sensitive PLL circuitry..
> As the SG48 got only a single GND at its thermal pad the pcb designers are in doubt where to wire the decoupling capacitors, it seems.. -- https://www.eevblog.com/forum/microcontrollers/8$-ice40-developer-board/msg1396333/#msg1396333

**Reply by josuah** [reply]
# D1

![2022-08-19-120028_447x362_scrot](https://user-images.githubusercontent.com/55351021/185595229-4d04e6c3-b8bb-439e-808b-124bb90cd132.png)

**Q: I cannot see why it is current would be let flowing between these two power rails: would that not heat-up the device to have a short there?**

Although, not much current: between 1mA and 10mA if I read it right:

![2022-08-19-121856_924x721_scrot](https://user-images.githubusercontent.com/55351021/185599669-603d8233-7220-42e1-b5fa-e720a84db944.png)
-- https://www.vishay.com/docs/85748/1n4148w.pdf

**Reply by josuah** [reply]
# C11 and C6 (solved)

Hardly any closer to the RP2040
![2022-08-19-123415_898x886_scrot](https://user-images.githubusercontent.com/55351021/185600732-0f7187d0-b449-4232-ad56-e0725ad98136.png)

Could be made 4.7uF for reducing the BOM, deviating only slightly from the recommendation, but for the main input power, probably good to use slightly higher values.

![2022-08-19-123658_1200x812_scrot](https://user-images.githubusercontent.com/55351021/185601105-c0c780d1-b6e1-4a06-8e8f-32365add3f71.png)
-- https://datasheets.raspberrypi.com/rp2040/hardware-design-with-rp2040.pdf#page=7

**Reply by josuah** [reply]
# C1 and C2 and R1 (solved)

Could be made 0.1uF (100pF) instead of 27pF, but here it is not about stability, but filtering for a crystal. Probably better let as recommended.

![2022-08-19-124035_662x254_scrot](https://user-images.githubusercontent.com/55351021/185601639-5d8db20f-d830-4cdc-bd54-4e93dc5869a1.png)

R1 is 1k, but also part of that filter.

**Reply by josuah** [reply]
# R2 (solved)

![2022-08-19-125847_953x606_scrot](https://user-images.githubusercontent.com/55351021/185604429-3333e02a-0eb6-4651-a4e5-1631a8a29a89.png)

R2 is 1k, same as R1 and reason is expained in https://datasheets.raspberrypi.com/rp2040/hardware-design-with-rp2040.pdf#page=9 :

![2022-08-19-125521_557x377_scrot](https://user-images.githubusercontent.com/55351021/185603969-b090dd59-fa8d-4ff7-ba3c-9c5ab32cf348.png)

> R2 is marked as DNF (Do Not Fit) on the schematic, as we have found that with this particular flash device, the external pull-up is unnecessary. However, if a different flash is used, it may become important to be able to insert a 10kΩ resistor here, so it has been included just in case.

**Comment by josuah** [comment]
**15-pin headers vs 16-pin header**
**@e2k** said on the [chat server](https://discord.com/channels/644405956494753803/985342148884987934):
> probably a stupid remark, but I always try to use connectors of "standard"sizes ... You used 15x2, 16x2 is easier to get ... I know, you can cut the yourself, but than again, the 32x2 is a standard ... (by standard I mean, that there is usually more of them in stock)

**Reply by josuah** [reply]
I think these are two regular 6x2 (as used in PMOD) headers, as well as one 3x2 header in-between the two PMODs

![2022-08-19-131348_879x510_scrot](https://user-images.githubusercontent.com/55351021/185607084-60086fab-2143-4ee8-9a7b-ccdc8961d7a0.png)

Which would help with figuring out where each PMOD starts: by having discontinuity in the header plastic acting as markings.

Visually, it is more obvious that the PMODs are at the left and right
here `|_______|___|_______|` than
here `|___________________|`

**Comment by josuah** [comment]
**NCP1117ST33T3G** is out of stock on Digikey and Mouser**
https://www.digikey.com/en/products/detail/onsemi/ncp1117st33t3g/1749063
https://www.mouser.fr/ProductDetail/onsemi/NCP1117ST33T3G

But present at multiple other providers : https://octopart.com/search?q=ncp1117st33t3g&currency=USD&specs=0

Not sure that is any concern, it uses a standard SOT-223 footprint and there will be plenty of drop-in replacements if needed : https://www.digikey.com/en/products/filter/pmic/pmic-voltage-regulators-linear/699?s=N4IgjCBcpgbFoDGUBmBDANgZwKYBoQB7KAbRABYAGAVgGYAOAdhAF0CAHAFyhAGVOATgEsAdgHMQAXwLkI0EMkjps%2BIqRC1yjcrQhsQXHv2HipBMI0YBOBAtSZcBYpDIXZYaqw7dIfQaIlpcCt6G3lFZUc1FxBKViCAWgAmW0VBAFdVZzJPFkSwpCgMrPU9SXKgA

**Reply by josuah** [reply]
These results from DigiKey max out at 500mA though
